webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+        meterRegistry.counter(
+            "AITranslateJob.noPendingMTFound", Tags.of("repository", repository.getName()));
+      }
+    } catch (Exception e) {
+      logger.error("Error running job for text unit id {}", input.getTmTextUnitId(), e);
+      meterRegistry.counter(
+          "AITranslateJob.error", Tags.of("repository", input.getRepositoryId().toString()));
+    } finally {
+      TmTextUnitPendingMT pendingMT =
+          tmTextUnitPendingMTRepository.findByTmTextUnitId(input.getTmTextUnitId());
+      if (pendingMT != null) {
+        logger.debug("Deleting pending MT for tmTextUnitId: {}", input.getTmTextUnitId());
+        tmTextUnitPendingMTRepository.delete(pendingMT);
+      }
+    }
+    return null;
+  }
+
+  private void translateLocales(
+      AITranslateJobInput input, TMTextUnit tmTextUnit, Repository repository) {
+
+    Map<String, RepositoryLocaleAIPrompt> repositoryLocaleAIPrompts =
+        repositoryLocaleAIPromptRepository
+            .getActiveTranslationPromptsByRepository(input.getRepositoryId())
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    rlap ->
+                        rlap.getLocale() != null
+                            ? rlap.getLocale().getBcp47Tag()
+                            : REPOSITORY_DEFAULT_PROMPT,
+                    Function.identity()));
+    repository.getRepositoryLocales().stream()
+        .filter(rl -> rl.getParentLocale() != null)
+        .map(RepositoryLocale::getLocale)
+        .forEach(
+            targetLocale -> {
+              try {
+                if (isExistingVariant(tmTextUnit, targetLocale)) {
+                  // If a translation already exists, skip the AI translation (likely exists due to
+                  // leveraging)
+                  logger.debug(
+                      "Translation already available for text unit id {} and locale: {}, skipping AI translation.",
+                      tmTextUnit.getId(),
+                      targetLocale.getBcp47Tag());
+                  meterRegistry.counter(
+                      "AITranslateJob.translate.alreadyExists",
+                      Tags.of(
+                          "repository",
+                          repository.getName(),
+                          "locale",
+                          targetLocale.getBcp47Tag()));
+                  return;
+                }
+                String sourceLang = repository.getSourceLocale().getBcp47Tag().split("-")[0];
+                if (isReuseSourceOnLanguageMatch
+                    && targetLocale.getBcp47Tag().startsWith(sourceLang)) {
+                  reuseSourceStringAsTranslation(tmTextUnit, repository, targetLocale, sourceLang);
+                  return;
+                }
+                // Get the prompt override for this locale if it exists, otherwise use the
+                // repository default
+                RepositoryLocaleAIPrompt repositoryLocaleAIPrompt =
```

Comment:
Do we want/need to track which prompt ID was used (default vs. language specific) for specific TUs and prompt executions?
Would it be too much noise in the logging layer do you think?